### ISSUE TITLE
conf-libcurl does not install on OS X

### ISSUE THREAD

**Issue by talex5** [ocaml/opam-repository#7347]
```
$ opam install ocurl
The following actions will be performed:
  ∗  install conf-libcurl 1                   [required by ocurl]
  ∗  install ocurl        0.7.7
===== ∗  2 =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[ocurl] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[ERROR] The compilation of conf-libcurl failed at "pkg-config libcurl".

#=== ERROR while installing conf-libcurl.1 ====================================#
# opam-version 1.2.2
# os           darwin
# command      pkg-config libcurl
# path         /Users/tal/.opam/system/build/conf-libcurl.1
# compiler     system (4.03.0)
# exit-code    1
# env-file     /Users/tal/.opam/system/build/conf-libcurl.1/conf-libcurl-50065-58c514.env
# stdout-file  /Users/tal/.opam/system/build/conf-libcurl.1/conf-libcurl-50065-58c514.out
# stderr-file  /Users/tal/.opam/system/build/conf-libcurl.1/conf-libcurl-50065-58c514.err



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
The following actions were aborted
  ∗  install ocurl 0.7.7
The following actions failed
  ∗  install conf-libcurl 1
No changes have been performed

=-=- conf-libcurl.1 troobleshooting -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
=> This package relies on external (system) dependencies that may be missing. `opam depext
   conf-libcurl.1' may help you find the correct installation for your system.
```

The suggested `depext` command does nothing.

However, it does work if I first do:

```
export PKG_CONFIG_PATH=/usr/local/Library/Homebrew/os/mac/pkgconfig/10.11:/usr/lib/pkgconfig
```

**Comment by faerietree** [comment]
@talex5 : first, epic 0install! Now I have the same /problem like you had on MacOS here on voidlinux trying to install dependencies using opam when compiliing 0install:
conf-libcurl.1 failed
The same for conf-libm4.1 and the troubleshooting via `opam depext
   conf-libcurl.1` just produces:
`No extra OS packages requirements found.` 

`export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig`
prior to the install command:
`opam install yojson xmlm ounit react lwt extlib ocurl obus lablgtk sha`
does not help  either.

Maybe this can help OPAM developers hunt down the issue. I miss 0install on epic voidlinux. :|

**Comment by faerietree** [comment]
*updated above post with how to reproduce information*

**Comment by faerietree** [comment]
New insights:
The m4 failed due to 
    + + echo
    m4
    sh: 1: m4: not found
according to the stderr file `~/.opam/system/build/conf-m4.1-28171-d8a4ae.err`

**Comment by faerietree** [comment]
Installing m4 manually using `xbps-install -Syv m4` resolved the m4 issue.

Still trying to fix the libcurl issue ...

**Comment by faerietree** [comment]
https://curl.haxx.se/mail/lib-2016-01/0055.html says to direct the PKG_CONFIG_PATH to the directory where the correct openssl.pc is located.

Because I compiled openssl from source as the voidlinux delivered libreSSL is not compatible / used by all applications, I therefore pointed it to ~/openssl/ where I compiled openssl:
`export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig:/home/anon/openssl/`

Unfortunately the error persists.

**Comment by faerietree** [comment]
Checked the openssl.pc files are identical. They are.

As a workaround I installed curl manually and only installed the basic requirements using opam that are not so easy to install via xbps-install because such packages not exist for voidlinux. Compiling from source is an option, yet OPAM is way quicker here and now that the m4 issue is resolved the following packages install flawlessly using OPAM:

`opam install yojson xmlm ounit react lwt extlib sha`

**Comment by faerietree** [comment]
As Thomas pointed out too, the `opam depext` does not help for some distributions / operating systems. Though according to 
https://opam.ocaml.org/doc/FAQ.html#Somepackagefailduringcompilationcomplainingaboutmissingdependenciesquotm4quotquotlibgtkquotetc
there also is `opam list --rec --required-by ocurl,ocamlfind --external` here using ocurl, ocamlfind packages as examples. This hints us to libcurl package which is installed.

So no progress getting ocurl to install using OPAM.

**Comment by faerietree** [comment]
`opam list --rec --required-by conf-libcurl.1 --external`
pointed to the development headers of libcurl being required.

After adding these using `xbps-install -Syv libcurl-devel` still `opam install ocurl` fails.

**Comment by faerietree** [comment]
My newest guess is that the development headers are required.
So I figured here on voidlinux the headers are in `/usr/local/include/openssl`.
Tried to provide them to opam / ocaml using (tell me if that's the wrong way to do it):
`OPENSSL_DEV=/usr/local/include/openssl && export EXTRA_CFLAGS=-I$OPENSSL_DEV && export EXTRA_CXXFLAGS=-I$OPENSSL_DEV && export EXTRA_LDFLAGS=-L/usr/local/lib`

Running `opam install ocurl` still results in conf-libcurl.1 not finding external libssl-devel headers it seems. (same output as ever before, conf-libcurl.1 troubleshooting)

**Comment by faerietree** [comment]
Denser evindence for libssl-dev being an issue:
voidlinux uses LibreSSL.
Custom compilation of openSSL somehow is not properly found or only partly.

Another install of pybloomfilter resulted in exit code 1 failure which may indicate missing libssl-devel according to http://askubuntu.com/questions/779803/installing-pybloomfiltermmap-version-0-3-14-via-pip ...

May it be I regret that LibreSSL and OpenSSL forked?

**Comment by faerietree** [comment]
At least one thing is positive: All this shows why 0install is so epic! Holy cow how I miss it. Vicious cycle though because I can not compile it which is why I post here. Teutates and Brexit, what else is to come!

**Comment by faerietree** [comment]
Easy fix on voidlinux:
`xbps-install -Syv pkg-config`
credit to @duncaen on voidlinux.xbps.irc.
also check that `libressl-devel` is installed.

**Comment by dbuenzli** [comment]
The original problem has been solved by @dsheets in b434de7dad48cb60f996e05ba1c63678bc290e8a.

@faerietree it was difficult for me to understand what your problem was exactly. If you are still having particular issues do not hesitate to open a new one with exact details.
